docs/concepts/types.md

```diff
@@ -122,6 +122,49 @@ except ValidationError as exc:
     """
 ```
 
+#### Experimental pipeline API
+
+Pydantic v2.8.0 introduced an experimental pipeline API that allows composing parsing, constraints and transformations in a more type-safe manner than existing APIs. This API is subject to change or removal, we are looking for feedback and suggestions before making it a permanent part of Pydantic.
+
+```python
+from __future__ import annotations
+
+from datetime import datetime
+
+from typing_extensions import Annotated
+
+from pydantic import BaseModel
+from pydantic.transform_experimental import parse, parse_defer
```

Comment:
Bikeshed on `parse`, in general I can't see many better choices, though I do quite like `take` or `takes` as a possibility. I also considered `val`. I can't see much else.

My nit against `parse` is I think it implies text as the input, which isn't true if the inputs are Python objects.

Reply:
I can live with `parse` if everyone else is also fine with it 👍